pkg/core/annotation.go

```diff
@@ -42,7 +45,7 @@ func (a *Annotation) MarshalJSON() ([]byte, error) {
 	return json.Marshal(m)
 }
 
-func (a Annotation) Format(s fmt.State, verb rune) {
+func (a *Annotation) Format(s fmt.State, verb rune) {
```

Comment:
Why change these to pointer? The compiler will automatically reference/dereference as needed so this shouldn't be necessary. It is a read-only function so does not need a pointer receiver.

Reply:
It's just an idiomatic go thing. Mixing pointer and value receivers is bad practice and can mess with a struct's ability to implement interfaces.

Reply:
> can mess with a struct's ability to implement interfaces.

In this case, it was intentionally not a pointer so the following would work properly:
```go
fmt.Sprintf("%+v", Annotation{...})
fmt.Sprintf("%+v", &Annotation{...}) // this works as well since the compiler will automatically dereference
```